spring-batch-docs/asciidoc/readersAndWriters.adoc

```diff
@@ -2779,221 +2779,255 @@ reconstituted from the execution context if the stream was
-to merge any Entities that aren't part of the persistence context.
+===== `JpaItemWriter`
+The `JpaItemWriter` is an `ItemWriter` that uses a JPA `EntityManagerFactory`
+to merge any entities that are not part of the persistence context.
 Spring Batch provides a `JpaItemWriterBuilder` to construct an instance
 of the `JpaItemWriter`.
 
 [[specializedReaders]]
 ==== Specialized Readers
-Here are some specialized readers offered by Spring Batch out of the box.
+Spring Batch offers the following specialized readers:
+
+* <<ldifReader>>
+* <<mappingLdifReader>>
 
 [[ldifReader]]
-===== LdifReader
-The `LdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them
+===== `LdifReader`
+The `LdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them,
 and returns a `LdapAttribute` object for each `read` executed.
 Spring Batch provides a `LdifReaderBuilder` to construct an instance of the `LdifReader`.
 
 
 [[mappingLdifReader]]
-===== MappingLdifReader
+===== `MappingLdifReader`
 The `MappingLdifReader` reads LDIF (LDAP Data Interchange Format) records from a `Resource`, parses them
 then maps each LDIF record to a POJO (Plain Old Java Object). Each read returns a POJO.
 Spring Batch provides a `MappingLdifReaderBuilder` to construct an instance of the `MappingLdifReader`.
 
 [[specializedWriters]]
 ==== Specialized Writers
-Here are some specialized writers offered by Spring Batch out of the box.
+Spring Batch offers the following specialized writers:
+
+* <<classifierCompositeItemWriter>>
+* <<simpleMailMessageItemWriter>>
+* <<gemfireItemWriter>>
 
 [[classifierCompositeItemWriter]]
-===== ClassifierCompositeItemWriter
-The `ClassifierCompositeItemWriter` calls one of a collection of ItemWriters for
+===== `ClassifierCompositeItemWriter`
+The `ClassifierCompositeItemWriter` calls one of a collection of `ItemWriter` implementations for
 each item, based on a router pattern implemented through the provided
 `Classifier`.  The implementation is thread-safe if all delegates are thread-safe.
 Spring Batch provides a `ClassifierCompositeItemWriterBuilder` to construct an
 instance of the `ClassifierCompositeItemWriter`.
 
 [[simpleMailMessageItemWriter]]
-===== SimpleMailMessageItemWriter
-The SimpleMailMessageItemWriter is an `ItemWriter` that can send mail messages.
+===== `SimpleMailMessageItemWriter`
+The `SimpleMailMessageItemWriter` is an `ItemWriter` that can send mail messages.
 It delegates the actual sending of messages to an instance of `MailSender`.
 Spring Batch provides a `SimpleMailMessageItemWriterBuilder` to construct an
 instance of the `SimpleMailMessageItemWriter`.
 
 
 [[gemfireItemWriter]]
-===== GemfireItemWriter
-The GemfireItemWriter is an `ItemWriter` that uses an `GemfireTemplate` that
+===== `GemfireItemWriter`
+The `GemfireItemWriter` is an `ItemWriter` that uses a `GemfireTemplate` that
```

Comment:
Gemfire should be under the other database writers like Neo4j...

Reply:
Again, moving this section involves moving the section, the anchor, and the link.